libs/vendored/jwtool/src/lib/jwtool.spec.ts

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as http from 'http';
+import { JWTool, PrivateJWK } from '../index';
+
+describe('JWTool', () => {
+  let server: http.Server;
+  let trustedJKUs: any;
+  let secretJWK: PrivateJWK;
+  const msg = { bar: 'baz' };
+
+  beforeAll(async () => {
+    server = http.createServer(function (req, res) {
+      res.writeHead(200, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify({ keys: [publicJWK] }));
+    });
+    const isListening = new Promise((resolve) =>
+      server.once('listening', resolve)
+    );
+    server.listen(0, '127.0.0.1');
+    await isListening;
+    const serverAddress = server.address();
+    if (!serverAddress) throw new Error('Server address not available');
+    const port =
+      typeof serverAddress === 'string' ? serverAddress : serverAddress.port;
```

Comment:
Converting the original JS test to TS highlighted this issue (which is a lovely thing about TS!), this was definitely a TIL for me as well.

Reply:
After reading this a bit more and digging into the Node docs for named Unix pipes, I don't _think_ this is correct. IIUC a named pipe is a semi-permanent connection by name/string, so it's probably returning the link to the connection file. But we're interpreting it as a port number which seems a bit suspicious.
Not sure how I would be able to test that theory locally, but also not sure it matters at all since we're listening on port `0` (which will give us a randomly available port number) and we aren't using pipes anyways.